test/geometry_tests.cpp

```diff
@@ -296,6 +298,89 @@ TEST(raytrace, negative_length) {
 	EXPECT_EQ(pixels.size(), 0);	
 }
 
+struct PolygonScenario{
+	std::vector<Cell> vertices;
+	int size_x;
+	int expected_size;
+	std::vector<Cell> expected_elements;
+};
+
+class polygonOutlineCellsTests : public :: testing::TestWithParam<PolygonScenario> {};
+
+TEST_P(polygonOutlineCellsTests, outline_tests){
+
+	auto curr_scenario = GetParam();
+
+	// GIVEN a the vertices of a small square, and a large size
+	const std::vector<Cell> vertices = curr_scenario.vertices;
+	const auto size_x = curr_scenario.size_x;
+
+	// WHEN the outline is created
+	const auto outline = polygonOutlineCells(vertices, size_x);
+
+	// THEN the resulting outline should be 20 elements and equal to expected 
+	const auto expected_size = curr_scenario.expected_size;
+	const auto expected_elements = curr_scenario.expected_elements;
+
+	EXPECT_EQ(outline.size(), expected_size);
+	EXPECT_THAT(outline, ElementsAreArray(expected_elements));		
+
+}
+
+INSTANTIATE_TEST_CASE_P(
+	polygonOutlineCells_tests,
```

Comment:
:stuck_out_tongue_closed_eyes: 
I usually write `PolygonOutlineCellsGroup` since that's what they give in the examples.

Reply:
Changed